File: Java/ex6.java
class Motor {
    private int cilindradas;
    private String tipoCombustivel;

    public Motor(int cilindradas, String tipoCombustivel) {
        this.cilindradas = cilindradas;
        this.tipoCombustivel = tipoCombustivel;
    }

    public String detalhesMotor() {
        return cilindradas + "cc, Combustível: " + tipoCombustivel;
    }
}

class Carro {
    private String marca;
    private String modelo;
    private int ano;
    private Motor motor;

    public Carro(String marca, String modelo, int ano, Motor motor) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.motor = motor;
    }

    public String detalhes() {
        return "Marca: " + marca + ", Modelo: " + modelo + ", Ano: " + ano + ", Motor: " + motor.detalhesMotor();
    }
}

public class Main {
    public static void main(String[] args) {
        Motor motorTeste = new Motor(2000, "Gasolina");
        Carro carroTeste = new Carro("Honda", "Civic", 2020, motorTeste);
        System.out.println(carroTeste.detalhes());
    }
}

File: Golang/ex6.go
package main

import "fmt"

type Motor struct {
	cilindradas     int
	tipoCombustivel string
}

func (m Motor) DetalhesMotor() string {
	return fmt.Sprintf("%dcc, Combustível: %s", m.cilindradas, m.tipoCombustivel)
}

type Carro struct {
	marca  string
	modelo string
	ano    int
	motor  Motor
}

func (c Carro) Detalhes() string {
	return fmt.Sprintf("Marca: %s, Modelo: %s, Ano: %d, Motor: %s", c.marca, c.modelo, c.ano, c.motor.DetalhesMotor())
}

func main() {
	motorTeste := Motor{2000, "Gasolina"}
	carroTeste := Carro{"Honda", "Civic", 2020, motorTeste}
	fmt.Println(carroTeste.Detalhes())
}
